test/unit/fact_cleaner_test.rb

```diff
@@ -48,6 +48,19 @@ class FactCleanerTest < ActiveSupport::TestCase
     assert_include remaining_of(root_6_fact), root_6_fact
   end
 
+  test "it cleans excluded facts" do
+    cleaner.stubs(:excluded_facts).returns(['ignored*', '*bad'])
+    cleaner.stubs(:delete_orphaned_facts).returns(0)
```

Comment:
This method (`delete_orphaned_facts`) needs a test.

Reply:
`delete_orphaned_facts` was extracted out of `clean!`. All other tests are actually testing it.